common/txmgr/types/attempt_builder.go

```diff
@@ -0,0 +1,45 @@
+package types
+
+import (
+	"context"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services"
+)
+
+// AttemptBuilder takes the base unsigned transaction + optional parameters (tx type, gas parameters)
+// and returns a signed TxAttempt
+// it is able to estimate fees and sign transactions
+// H - chain head type
+// F - chain fee type
+// A - chain address type
+// HA - chain hash type
+// GU - chain gas unit type
+// T - tx type (will be replaced in future)
+// TA - tx attempt type (will be replaced  in future)
+//
+//go:generate mockery --quiet --name AttemptBuilder --output ./mocks/ --case=underscore
+type AttemptBuilder[H Head, F Fee, A any, HA any, GU any, T any, TA any] interface {
+	// interfaces for running the underlying estimator
+	services.ServiceCtx
+	HeadTrackable[H]
+
+	// NewAttempt builds a transaction using the configured transaction type and fee estimator (new estimation)
+	NewAttempt(ctx context.Context, etx T, lggr logger.Logger, opts ...Opt) (attempt TA, fee F, feeLimit uint32, retryable bool, err error)
+
+	// NewAttemptWithType builds a transaction using the configured fee estimator (new estimation) + passed in tx type
+	NewAttemptWithType(ctx context.Context, etx T, lggr logger.Logger, txType int, opts ...Opt) (attempt TA, fee F, feeLimit uint32, retryable bool, err error)
+
+	// NewBumpAttempt builds a transaction using the configured fee estimator (bumping) + passed in tx type
+	// this should only be used after an initial attempt has been broadcast and the underlying gas estimator only needs to bump the fee
+	NewBumpAttempt(ctx context.Context, etx T, previousAttempt TA, txType int, priorAttempts []PriorAttempt[F, HA], lggr logger.Logger) (attempt TA, bumpedFee F, bumpedFeeLimit uint32, retryable bool, err error)
+
+	// NewCustomAttempt builds a transaction using the passed in fee + tx type
+	NewCustomAttempt(etx T, fee F, gasLimit uint32, txType int, lggr logger.Logger) (attempt TA, retryable bool, err error)
```

Comment:
Could we rename to NewExplicitFeeAttempt?
To indicate it is used if caller wants to set their own explicit fee.
Also, can we remove the txType from this? Can it be derived from config?
Because if we can, then we get rid of the TxType from the general TXM. The TxType is really an artifact of EIP1559, and I don't want it to leak outside.

Reply:
^ something like `NewAttemptCustomFee` works

Reply:
i'm happy to rename the func :)

I would actually prefer leaving `TxType` in, it's not just an artifact of EIP1559 - but also solana has different transaction types, even other EVM chains like klaytn have different tx types. So having knowledge of the specific TxType is useful